### ISSUE TITLE
Filesystem routing doesn't respect routing precedences

### ISSUE THREAD

**Issue by devunt** [brillout/vite-plugin-ssr#448]
### Description

Reproduction repo: https://github.com/devunt/vite-plugin-ssr-routing-bug
Codesandbox: https://codesandbox.io/p/github/devunt/vite-plugin-ssr-routing-bug/draft/practical-mestorf

Filesystem routing does not respect rules of routing precedences. Quote from https://vite-plugin-ssr.com/routing-precedence indicates that:
> Upon Route String conflicts, vite-plugin-ssr chooses the first route from most specific to least specific

In the reproduction repo, I have following directory structure:
```
/index.page.tsx <-- works well (obivously)

/1/index.page.tsx <-- works well
/1/@param.page.tsx <-- every routes including /1/specific falls to here
/1/specific.page.tsx <-- I cannot access to this page

/2/index.page.tsx <-- works well
/2/@param.page.tsx <-- every routes including /2/deep falls to here
/2/deep/index.page.tsx <-- I cannot access to this page
```

It seems "most specific to least specific" rule doesn't apply to filesystem routing.
I think it should be applied in filesystem routing, too?

**P.S.**
I sponserd @brillout on the behalf of @ccoli-co, a company I and @XiNiHa work for.
Thank you for making and maintaining great plugin 🎉

We are in the middle of progress of migrating existing Next.js site (which have over hundred pages) to VPS, and it is being a very pleasant experience in both DX and performance. As now VPS is being crucial part of our business and frontend stack, we'd made an choice to sponser its maintainer and ecosystem.

**Comment by brillout** [comment]
Yea, I agree. Let me have a look.

Thanks a lot for your sponsorship... 💚. It makes a tremendous difference. I'm humbled. Thanks!

**Comment by brillout** [comment]
Fix released in `0.4.34`. I didn't test the fix; let me know if you still run into any issue.
